docs/content/contributing/terraform/terraform-contribution-flow/_index.md

```diff
@@ -0,0 +1,316 @@
+G(<a href='/Azure-Verified-Modules/contributing/terraform/terraform-contribution-flow/#7-create-a-pull-request-to-the-upstream-repository'>7. Create a pull request to the upstream repository</a>)
+A --> B
+B --> C
+C --> D
+D --> E
+F -->|yes|G
+F -->|no|E
+{{< /mermaid >}}
+
+<br>
+
+---
+
+<br>
+
+## GitFlow for contributors
+
+The GitFlow process outlined here depicts and suggests a way of working with Git and GitHub. It serves to synchronize the forked repository with the original upstream repository. It is not a strict requirement to follow this process, but it is highly recommended to do so.
+
+{{< mermaid class="text-center" >}}
+%%{init: { 'logLevel': 'debug', 'gitGraph': {'rotateCommitLabel': false}} }%%
+gitGraph LR:
+commit id:"fork"
+branch fork/main
+checkout fork/main
+commit id:"checkout feature" type: HIGHLIGHT
+branch feature
+checkout feature
+commit id:"checkout fix"
+branch fix
+checkout main
+merge feature id: "Pull Request 'Feature'" type: HIGHLIGHT
+checkout fix
+commit id:"Patch 1"
+commit id:"Patch 2"
+checkout main
+merge fix id: "Pull Request 'Fix'" type: HIGHLIGHT
+{{< /mermaid >}}
+
+{{< hint type=tip >}}
+
+When implementing the GitFlow process as described, it is advisable to configure the local clone of your forked repository with an additional remote for the upstream repository. This will allow you to easily synchronize your locally forked repository with the upstream repository. **_Remember_**, there is a difference between the forked repository on GitHub and the clone of the forked repository on your local machine.
+
+<img src="/Azure-Verified-Modules/img/contribution/forkUpstream.png" alt="Upstream to fork and source repository." width=50%>
+
+{{< /hint >}}
+
+<br>
+
+---
+
+<br>
+
+<!--
+TODO:
+
+Checklist
+
+Run grept
+Update provider versions
+cd root
+avmfix -folder . # examples too
+make fmt
+make docs
+
+-->
+
+{{< hint type=note >}}
+
+Each time in the following sections we refer to 'your xzy', it is an indicator that you have to change something in your own environment.
+
+{{< /hint >}}
+
+## Prepare your developer environment
+
+### 1. Fork the module source repository
+
+{{< hint type=important >}}
+
+Each Terraform AVM module will have its own GitHub repository in the [`Azure`](https://github.com/Azure) GitHub Organization as per [SNFR19](/Azure-Verified-Modules/specs/shared/#id-snfr19---category-publishing---registries-targeted).
+
+This repository will be created by the Module owners and the AVM Core team collaboratively, including the configuration of permissions as per [SNFR9](/Azure-Verified-Modules/specs/shared/#id-snfr9---category-contributionsupport---avm--pg-teams-github-repo-permissions)
+
+{{< /hint >}}
+
+Module contributors are expected to fork the corresponding repository and work on a branch from within their fork, before then creating a Pull Request (PR) back into the source repository's `main` branch.
```

Comment:
Not if they are a member of the contributors group

Reply:
This stands in relation tho the above mentioned comment about outside collaborators (non-Azure-org). Once we found consensus happy to adjust the order and adjust the wording. Marking this as `TBD`.